### PR TITLE
Move the 'Update' and 'Install' buttons to the app bar

### PR DESCRIPTION
**Pull request by janfaracik**
https://github.com/jenkinsci/jenkins/assets/43062514/b419e317-941d-4cd6-96c6-e0ff3412b097

image

This PR removes the bottom app bar from the Plugins pages, moving the final control to the top app bar. This increases the available screen real estate for updates. The 'Update' and 'Install' buttons will now be disabled unless a plugin is checked, whereas previously you could click it and it would do nothing.

### Testing done

* Plugins install/update as before
* The 'Update' and 'Install' buttons enable/disable depending on if an update is checked

### Proposed changelog entries

- Move the 'Update' and 'Install' buttons to the app bar.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8025"></a>

### CODE REVIEW

File: core/src/main/resources/hudson/PluginManager/updates.jelly
Comment (StefanSpieker) [inline core/src/main/resources/hudson/PluginManager/updates.jelly:240]: Is this functionally removed?

**Review by StefanSpieker — COMMENTED** [review]
I really like this, I'm just not sure if the download and install directly functionally is still needed for updates. In the install tab it is still present?

**Comment by mawinter69** [comment]
The change only affects the plugin updates page
What about the available plugins page? It would be inconsistent to have the buttons there still at the bottom

**Comment by janfaracik** [comment]
> The change only affects the plugin updates page
> What about the available plugins page? It would be inconsistent to have the buttons there still at the bottom

Happy to make that change as part of this PR or a quick follow up PR. Any thoughts @here?

**Comment by timja** [comment]
> > The change only affects the plugin updates page
> > What about the available plugins page? It would be inconsistent to have the buttons there still at the bottom
> 
> Happy to make that change as part of this PR or a quick follow up PR. Any thoughts?

the PR is quite small unless updating that will change a lot I think better to do the plugin manager as a whole

**Comment by janfaracik** [comment]
I've updated it to include the buttons - 

image

It's a dropdown button (due to a lack of space on smaller screens). Happy to change it into a [split button](https://www.nngroup.com/articles/split-buttons/) if people prefer.

**Comment by mawinter69** [comment]
I guess the majority of users select `Install without restart` so from that point of view a split button might be better.

**Comment by janfaracik** [comment]
> I guess the majority of users select `Install without restart` so from that point of view a split button might be better.

image

Updated to be a split button. Thanks for the feedback :)

**Comment by StefanSpieker** [comment]
That looks really nice, but the CI complains about prettier:

> [2023-06-10T19:22:42.471Z] [INFO] Running 'yarn lint:ci' in /home/jenkins/agent/workspace/Core_jenkins_PR-8025/war
> [2023-06-10T19:22:47.533Z] [INFO] Checking formatting...
> [2023-06-10T19:22:53.261Z] [INFO] [warn] src/main/js/plugin-manager-ui.js
> [2023-06-10T19:22:58.008Z] [INFO] [warn] Code style issues found in the above file. Forgot to run Prettier?

**Comment by NotMyFault** [comment]
> > I guess the majority of users select `Install without restart` so from that point of view a split button might be better.
> 
> image
> Updated to be a split button. Thanks for the feedback :)

Clicking on the button does nothing for me.

**Comment by timja** [comment]
Same I was just testing it and the button appears to be broken.

**Comment by timja** [comment]
(re-tested and looks great, waiting on ATH)

**Review by yaroslavafenkin — APPROVED** [review]
Looks fine security wise

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/hudson/PluginManager/available.jelly
Comment (jtnord) [inline core/src/main/resources/hudson/PluginManager/available.jelly:50]: using download icon to me is confusing as I (the user) am not downloading anything to my machine.
Rather Jenkins is installing - which is normally an arrow pointing right or the like.

**Pull request merged**
